osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -327,20 +328,15 @@ protected override void Update()
                 return;
 
             // Move the rectangle to cover the items
-            var topLeft = new Vector2(float.MaxValue, float.MaxValue);
-            var bottomRight = new Vector2(float.MinValue, float.MinValue);
+            RectangleF selectionRect = ToLocalSpace(selectedBlueprints.First().SelectionQuad).AABBFloat;
 
-            foreach (var blueprint in selectedBlueprints)
-            {
-                topLeft = Vector2.ComponentMin(topLeft, ToLocalSpace(blueprint.SelectionQuad.TopLeft));
-                bottomRight = Vector2.ComponentMax(bottomRight, ToLocalSpace(blueprint.SelectionQuad.BottomRight));
-            }
+            foreach (var blueprint in selectedBlueprints.Skip(1))
```

Comment:
Use indexed `for` loop rather than calling LINQ methods.

Reply:
ah right this is `Update()`, missed noticing that.

Reply:
Doesn't matter where it is, it just looks bad the way it's written here.  At the point you're accessing a list and using `.Skip` just to make `foreach` work, you probably shouldn't be using `foreach`.

Reply:
Aha, I was overly obsessed with LINQ to the point I use `.First()`/`.Last()` and `.Skip()`/`.Take()` regardless of whether the enumerable supports indexing or not, but I'll take this into account.